cmd/gateway/s3x/gateway-s3x-ledger-multipart.go

```diff
@@ -0,0 +1,170 @@
+
+/////////////////////
+// SETTER FUNCTINS //
+/////////////////////
+
+// AbortMultipartUpload is used to abort a multipart upload
+func (ls *ledgerStore) AbortMultipartUpload(bucket, multipartID string) error {
+	err := ls.AssertBucketExits(bucket)
+	if err != nil {
+		return err
+	}
+	return ls.DeleteMultipartID(multipartID)
+}
+
+// NewMultipartUpload is used to store the initial start of a multipart upload request
+func (ls *ledgerStore) NewMultipartUpload(multipartID string, info *ObjectInfo) error {
+	bucket := info.GetBucket()
+	defer ls.locker.write(bucket)
+	err := ls.assertBucketExits(bucket)
+	if err != nil {
+		return err
+	}
+
+	m := &MultipartUpload{
+		ObjectInfo: info,
+		Id:         multipartID,
+	}
+
+	ls.pmapLocker.Lock()
+	ls.l.MultipartUploads[multipartID] = m
+	ls.pmapLocker.Unlock()
+
+	data, err := m.Marshal()
+	if err != nil {
+		return err
+	}
+	return ls.ds.Put(dsPartKey.ChildString(multipartID), data)
+}
+
+// PutObjectPart is used to record an individual object part within a multipart upload
+func (ls *ledgerStore) PutObjectPart(bucketName, objectName, partHash, multipartID string, partNumber int64) error {
+	err := ls.AssertBucketExits(bucketName)
+	if err != nil {
+		return err
+	}
+
+	defer ls.plocker.write(multipartID)()
+	m, err := ls.getMultipartLoaded(multipartID)
+	if err != nil {
+		return err
+	}
+	m.ObjectParts[partNumber] = ObjectPartInfo{
+		Number:   partNumber,
+		DataHash: partHash,
+	}
+	data, err := m.Marshal()
+	if err != nil {
+		return err
+	}
+	return ls.ds.Put(dsPartKey.ChildString(multipartID), data)
+}
+
+/////////////////////
+// GETTER FUNCTINS //
+/////////////////////
+
+// GetObjectParts is used to return multipart upload parts,
+// returned unlock function must be used after map iteration is done.
+func (ls *ledgerStore) GetObjectDetails(id string) (*MultipartUpload, func(), error) {
+	unlock := ls.plocker.read(id)
+	m, err := ls.getMultipartLoaded(id)
+	if err != nil {
+		unlock()
+		return nil, nil, err
+	}
+	return m, unlock, nil
+}
+
+// MultipartIDExists is used to lookup if the given multipart id exists
+func (ls *ledgerStore) MultipartIDExists(id string) error {
+	return ls.accertValidUploadID(id)
+}
+
+// GetMultipartHashes returns the hashes of all multipart upload object parts
+/* not used for now
+func (ls *ledgerStore) GetMultipartHashes(bucket, multipartID string) ([]string, error) {
+	ex, err := ls.bucketExists(bucket)
+	if err != nil {
+		return nil, err
+	}
+	if !ex {
+		return nil, ErrLedgerBucketDoesNotExist
+	}
+	if err := ls.l.multipartExists(multipartID); err != nil {
+		return nil, err
+	}
+	mpart := ls.l.MultipartUploads[bucket]
+	var hashes = make([]string, len(mpart.ObjectParts))
+	for i, objpart := range mpart.ObjectParts {
+		hashes[i] = objpart.GetDataHash()
+	}
+	return hashes, nil
+}*/
+
+///////////////////////
+// INTERNAL FUNCTINS //
+///////////////////////
+
+// accertValidUploadID is a helper function to check if a multipart id exists in our ledger
```

Comment:
typo? should be `asssertValidUploadID`  i think

Reply:
thanks